### DISCUSSION TITLE
Sidebar items

### DISCUSSION THREAD

**Discussion by DennisSnijder** [pemedia/payload-visual-editor#7] [General]
As mentioned in the readme, the `position: sidebar` is not supported since the sidebar needs to make place for the visual editor.
What are the options for supporting `position: sidebar` fields in the future?

Perhaps a toggle where you can switch between the visual editor and the actual sidebar? Requiring you to switch back to the sidebar view to save / publish changes requiring you to fill any required `position: sidebar` fields.

What do you think?

**Comment by jpeiniger** [comment]
Thanks for addressing this.

What would be wonderful, would be to be able to move the admin-sidebar to a tab in the edit view. But a toggle to somehow get the info visible to be able to fill out the forms should be doable. I will play around a bit this week with the ideas. 

We already have to "x" to get rid of the preview, perhaps we can use this, to also bring back the admin-sidebar, as a temporary workaround.

**Reply by DennisSnijder** [reply]
Yes! that would be awesome!

**Reply by jpeiniger** [reply]
Perhaps we could really ad the sidebar fields manually via the custom UI field we also use for die visual editor itself.
It started looking in [Payloads edit views Default.tsx](https://github.com/payloadcms/payload/blob/51c2ab16728fc40c26be20ba861a6992c90da5f9/src/admin/components/views/collections/Edit/Default.tsx), but dunno yet if this will work as "easily" as I hope it does:

```
<RenderFields
  readOnly={!hasSavePermission}
  permissions={permissions.fields}
  filter={(field) => field?.admin?.position === 'sidebar'}
  fieldTypes={fieldTypes}
  fieldSchema={fields}
/>
  ```

**Reply by jpeiniger** [reply]
Here we go @DennisSnijder: https://github.com/pemedia/payload-visual-editor/pull/8

**Reply by DennisSnijder** [reply]
Nice! that works!

**Reply by jpeiniger** [reply]
Nice... it's released. I even think, I can move it to a separat tab.

**Reply by jpeiniger** [reply]
I couldn't resist. I pushed it into another branch. If you like, check it out:
https://github.com/pemedia/payload-visual-editor/tree/feat-admin-sidebar-tabs

The logic: If there are tabs present, the sidebar is added as a separate tab with the label "More" to the already existing tab group. If not, it behaves like the "old" version and gets attached to the main field area.

**Reply by DennisSnijder** [reply]
Nice! that makes it even beter IMO!

However the solutions still feels kinda "hidden". In the tab view it's just in another tab, with the currently merged solution the fields are always at the bottom of the view. Most of the time when I place something in a sidebar view, I want the fields to be there consistently. For example: I place a "publicationDate" field there, since the publish button is also over there.

Gotta admit it's pretty hard to come up with a good solution for this.... 😓

**Comment by jpeiniger** [comment]
From Payload 2.x on the sidebar issue is no longer relevant, as it is handled in core and also allowed us to make the plugin source much leaner and more focussed on the core functionality.
